tierkreis/tierkreis/worker/worker.py

```diff
@@ -121,6 +123,13 @@ def write_stubs(self, stubs_path: Path) -> None:
         with open(stubs_path, "w+") as fh:
             fh.write(format_namespace(self.namespace))
 
+        ruff_binary = shutil.which("ruff")
```

Comment:
https://github.com/astral-sh/ruff/blob/94947cbf65388988cbe926302831bc075547acd8/python/ruff/__main__.py#L8

Interestingly it seems like ruff does provide a method for finding ruff.